demosplan/DemosPlanCoreBundle/ResourceTypes/ProcedureMapSettingResourceType.php

```diff
@@ -0,0 +1,291 @@
+                });
+        }
+
+        if ($this->currentUser->hasPermission('area_procedure_adjustments_general_location')) {
+            $configBuilder->coordinate
+                ->updatable([], function (ProcedureSettings $procedureSettings, ?array $coordinate): array {
+                    $procedureSettings->setCoordinate($this->convertCoordinatesToFlatList($coordinate));
+
+                    return [];
+                })
+                ->readable(false, fn (ProcedureSettings $procedureSettings): ?array => $this->convertFlatListToCoordinates($procedureSettings->getCoordinate(), 2));
+        }
+
+        if ($this->currentUser->hasPermission('feature_map_use_territory')) {
+            $configBuilder->territory
+                ->updatable([], function (ProcedureSettings $procedureSettings, array $territory): array {
+                    $procedureSettings->setTerritory($this->convertCoordinatesToJson($territory));
+
+                    return [];
+                })
+                ->readable(false, fn (ProcedureSettings $procedureSettings): ?array => $this->convertJsonToCoordinates($procedureSettings->getTerritory()));
+        }
+
+        if ($this->currentUser->hasPermission('field_master_procedure_default_bounding_box')) {
+            $configBuilder->defaultBoundingBox
+                ->readable(false, function (ProcedureSettings $procedureSetting): ?array {
+                    $masterTemplateMapSetting = $this->masterTemplateService->getMasterTemplate()->getSettings();
+
+                    return $this->convertFlatListToCoordinates($masterTemplateMapSetting->getBoundingBox(), 4);
+                });
+        }
+
+        $configBuilder->defaultMapExtent
+            ->readable(false, function (ProcedureSettings $procedureSetting): ?array {
+                $masterTemplateMapSetting = $this->masterTemplateService->getMasterTemplate()->getSettings();
+
+                return $this->convertFlatListToCoordinates($masterTemplateMapSetting->getMapExtent(), 4);
+            });
+
+        $configBuilder->useGlobaInformationUrl
+            ->readable(false, fn (ProcedureSettings $procedureSetting): bool => $this->globalConfig->isMapGetFeatureInfoUrlGlobal());
+
+        return $configBuilder;
+    }
+
+    protected function getSetting(string $settingName, ProcedureSettings $procedureSetting): ?Setting
+    {
+        $settings = $this->contentService->getSettings(
+            $settingName,
+            SettingsFilter::whereProcedureId($procedureSetting->getProcedure()->getId())->lock(),
+            false);
+        Assert::countBetween($settings, 0, 1);
+
+        return array_pop($settings);
+    }
+
+    protected function convertStartEndCoordinatesToFlatList(?array $coordinates): string
+    {
+        return null === $coordinates ? '' : implode(',', [
+            $coordinates['start']['latitude'],
+            $coordinates['start']['longitude'],
```

Comment:
Constants may make sense here and for the other usages of these strings. But I don't know a good place or name either.